pufferlib/ocean/trash_pickup/trash_pickup.h

```diff
@@ -0,0 +1,596 @@
+    float* observations;
+    int* actions;
+    float* rewards;
+    unsigned char* dones;
+    LogBuffer* log_buffer;
+
+    int grid_size;
+    int num_agents;
+    int num_trash;
+    int num_bins;
+    int max_steps;
+    int current_step;
+
+    float positive_reward;
+    float negative_reward;
+    float total_episode_reward;
+
+    GridCell* grid; // 1D array for grid
+    int (*agent_positions)[2]; // Agent positions
+    int* agent_carrying; // Carrying status of agents
+    int (*trash_positions)[2];  // Positions of trash
+    int* trash_presence;        // Presence indicator for each piece of trash
+    int (*bin_positions)[2];    // Positions of trash bins
+
+    int obs_size;
+} CTrashPickupEnv;
+
+void compute_observations(CTrashPickupEnv* env) {
+    float* obs = env->observations;
+    float norm_factor = 1.0f / env->grid_size;
+
+    int obs_per_agent = env->obs_size / env->num_agents;
+
+    for (int agent_idx = 0; agent_idx < env->num_agents; agent_idx++) {
+        int obs_index = agent_idx * obs_per_agent;
+
+        // Add the observing agent's own position and carrying status
+        obs[obs_index++] = (float) (env->agent_positions[agent_idx][0]) * norm_factor;
+        obs[obs_index++] = (float) (env->agent_positions[agent_idx][1]) * norm_factor;
+        obs[obs_index++] = env->agent_carrying[agent_idx] ? 1.0f : 0.0f;
+
+        // Add positions of other agents
+        for (int other_idx = 0; other_idx < env->num_agents; other_idx++) {
+            if (other_idx == agent_idx) continue; // Skip self
+            obs[obs_index++] = (float) (env->agent_positions[other_idx][0]) * norm_factor;
+            obs[obs_index++] = (float) (env->agent_positions[other_idx][1]) * norm_factor;
+            obs[obs_index++] = env->agent_carrying[other_idx] ? 1.0f : 0.0f;
+        }
+
+        // Add shared trash data
+        for (int i = 0; i < env->num_trash; i++) {
+            if (env->trash_presence[i]) {
+                obs[obs_index++] = (float) (env->trash_positions[i][0]) * norm_factor;
+                obs[obs_index++] = (float) (env->trash_positions[i][1]) * norm_factor;
+                obs[obs_index++] = 1.0f; // Trash present
+            } else {
+                obs[obs_index++] = -1.0f; // Placeholder for x
+                obs[obs_index++] = -1.0f; // Placeholder for y
+                obs[obs_index++] = 0.0f; // Trash not present
+            }
+        }
+
+        // Add shared bin data
+        for (int i = 0; i < env->num_bins; i++) {
+            obs[obs_index++] = env->bin_positions[i][0] * norm_factor;
+            obs[obs_index++] = env->bin_positions[i][1] * norm_factor;
+        }
+    }
+}
+
+// Helper functions
+void place_random_items(CTrashPickupEnv* env, int count, int item_type) {
+    int placed = 0;
+    while (placed < count) 
+    {
+        int x = rand() % env->grid_size;
+        int y = rand() % env->grid_size;
+
+        if (env->grid[INDEX(env, x, y)].type == EMPTY) {
```

Comment:
Use inverted conditional pattern to reduce nesting. if !condition continue

Reply:
Fixed (to be pushed).